### DISCUSSION TITLE
Does github have an option for private scripts?

### DISCUSSION THREAD

**Discussion by Directinho** [community/community#22484] [New to GitHub]
<p><strong>I am creating a bot on Discord, I wanted to create a public repository on github for people and friends to follow my progress with the bot, but I was needing to create a private script to store more private things like the bot token, I am not finding this option and I don’t know if there is any way to create a file where only I can see it.</strong></p>
<p>Example</p>
<pre><code class="lang-auto">Index.js

Const Token = (./PrivateFile)
</code></pre>
<p><em>programs that I am using in creation:</em><br>
<em>Github = Save my codes and keep showing my progress</em><br>
<em>Visual Studio Code = Programming entirely from vscode</em><br>
<em>UptimeRobot and Replit = Host bot</em></p>

**Comment by kingthorin** [comment, marked as answer]
<p>Don’t use a file use repository secrets.</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/actions/security-guides/encrypted-secrets">
  <header class="source">
      <img src="https://docs.github.com/assets/cb-803/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/actions/security-guides/encrypted-secrets" target="_blank" rel="noopener nofollow ugc">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/7570458/181071393-3c74f01a-5c8d-43b2-b70d-4dfd896bef5b.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/actions/security-guides/encrypted-secrets" target="_blank" rel="noopener nofollow ugc">Encrypted secrets - GitHub Docs</a></h3>

  <p>Encrypted secrets allow you to store sensitive information in your organization, repository, or repository environments.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by Directinho** [comment]
<p>Oh, that’s what I needed, The value would be just my token or <code>"token"= "Token_here"</code></p>

**Comment by kingthorin** [comment]
<p>The value is the token, then you refer to it by the name of the secret.</p>
